application/src/main/java/org/opentripplanner/street/model/vertex/Vertex.java

```diff
@@ -35,6 +42,8 @@ public abstract class Vertex implements AStarVertex<State, Edge, Vertex>, Serial
 
   private transient Edge[] outgoing = new Edge[0];
   private RentalRestrictionExtension rentalRestrictions = RentalRestrictionExtension.NO_RESTRICTION;
+  private static final Set<AreaStop> EMPTY_SET = Set.of();
+  private Set<AreaStop> areaStops = EMPTY_SET;
```

Comment:
Very few Vertexes have areaStops, I am not an expert on the AStar routing and design, but this does not look right to me.